doc/specs/stdlib_linalg.md

```diff
@@ -600,6 +600,56 @@ Specifically, upper Hessenberg matrices satisfy `a_ij = 0` when `j < i-1`, and l
 {!example/linalg/example_is_hessenberg.f90!}
 ```
 
+## `solve` - Solves a linear matrix equation or a linear system of equations. 
+
+### Status
+
+Experimental
+
+### Description
+
+This function computes the solution to a linear matrix equation \( A \cdot x = b \), where \( A \) is a square, full-rank, `real` or `complex` matrix.
+
+Result vector or array `x` returns the exact solution to within numerical precision, provided that the matrix is not ill-conditioned. The solver is based on LAPACK's `*GESV` backends.
+
+### Syntax
+
+`Pure` interface:
+
+`x = ` [[stdlib_linalg(module):solve(interface)]] `(a, b)`
+
+Expert interface:
+
+`x = ` [[stdlib_linalg(module):solve(interface)]] `(a, b [, overwrite_a], err)`
+
+### Arguments
+
+Two 
+
+`a`: Shall be a rank-2 `real` or `complex` square array containing the coefficient matrix. It is normally an `intent(in)` argument. If `overwrite_a=.true.`, it is an `intent(inout)` argument and is destroyed by the call. 
+
+`b`: Shall be a rank-1 or rank-2 array of the same kind as `a`, containing the right-hand-side vector(s). It is an `intent(in)` argument.
+
+`overwrite_a` (optional): Shall be an input logical flag. if `.true.`, input matrix `a` will be used as temporary storage and overwritten. This avoids internal data allocation. This is an `intent(in)` argument.
+
+`err` (optional): Shall be a `type(linalg_state_type)` value. This is an `intent(out)` argument. The function is not `pure` if this argument is provided.
+
+### Return value
+
+Returns an array value that represents the solution to the linear system of equations.
+
```

Comment:
What happens if A is singular?

Reply:
`LINALG_ERROR` is raised: 

https://github.com/fortran-lang/stdlib/blob/316c44ad0a3a1868b9ead2596acdd6c4e4c9c231/doc/specs/stdlib_linalg.md?plain=1#L641

I've improved the documentation at [5e0620c](https://github.com/fortran-lang/stdlib/pull/806/commits/5e0620c3297f0d86e6517805c70786cf9061b52d), thanks @jvdp1